src/charm.py

```diff
@@ -237,6 +260,33 @@ def _loki_config(self) -> dict:
         )
         return yaml.safe_load(config)
 
+    @property
+    def _pod_ip(self) -> str:
+        """Returns the pod ip of this unit."""
+        if bind_address := self.model.get_binding(PEER).network.bind_address:
+            return str(bind_address)
+        return ""
+
+    @property
+    def _external_url(self) -> str:
+        """Return the external hostname to be passed to ingress via the relation."""
+        self.framework.breakpoint()
+        if "web_external_url" in self.model.config:
```

Comment:
This block must go :-)

Reply:
Iiuc, this condition is always True. Regardless, it is redundant since you have `if web_external_url := self.model.config["web_external_url"]`, which I would change to `if web_external_url := self.model.config.get("web_external_url")` just in case :D

Reply:
@Abuelodelanada yes